ktor-shared/ktor-serialization/ktor-serialization-tests/jvm/src/AbstractServerSerializationTest.kt

```diff
@@ -68,103 +68,95 @@ public abstract class AbstractServerSerializationTest {
+        client.get("/list").let { response -> verifyListResponse(response, Charsets.UTF_8) }
     }
 
     @Test
     public fun testListAcceptUtf16(): Unit = withTestSerializingApplication {
-        handleRequest(HttpMethod.Get, "/list") {
-            addHeader("Accept-Charset", "UTF-16")
-        }.let { call -> verifyListResponse(call.response, Charsets.UTF_16) }
+        client.get("/list") {
+            header(HttpHeaders.AcceptCharset, "UTF-16")
+        }.let { response -> verifyListResponse(response, Charsets.UTF_16) }
     }
 
     @Test
     public open fun testFlowNoAcceptUtf8(): Unit = withTestSerializingApplication {
-        handleRequest(HttpMethod.Get, "/flow").let { call ->
-            verifyListResponse(call.response, Charsets.UTF_8)
-            assertEquals("chunked", call.response.headers[HttpHeaders.TransferEncoding])
+        client.get("/flow").let { response ->
+            verifyListResponse(response, Charsets.UTF_8)
+            assertEquals("chunked", response.headers[HttpHeaders.TransferEncoding])
         }
     }
 
     @Test
     public open fun testFlowAcceptUtf16(): Unit = withTestSerializingApplication {
-        handleRequest(HttpMethod.Get, "/flow") {
-            addHeader("Accept-Charset", "UTF-16")
-        }.let { call -> verifyListResponse(call.response, Charsets.UTF_16) }
+        client.get("/flow") {
+            header(HttpHeaders.AcceptCharset, "UTF-16")
+        }.let { response -> verifyListResponse(response, Charsets.UTF_16) }
     }
 
-    private fun verifyListResponse(response: TestApplicationResponse, charset: Charset) {
-        assertEquals(HttpStatusCode.OK, response.status())
-        val bytes = response.byteContent
-        assertNotNull(bytes)
+    private suspend fun verifyListResponse(response: HttpResponse, charset: Charset) {
+        assertEquals(HttpStatusCode.OK, response.status)
+        val bytes = response.bodyAsBytes()
         val listEntity = simpleDeserializeList(bytes, charset)
         assertEquals(testEntities, listEntity)
     }
 
     @Test
     public open fun testMap(): Unit = withTestSerializingApplication {
-        handleRequest(HttpMethod.Get, "/map") {
-            addHeader("Accept", "application/json")
-        }.response.let { response ->
-            assertEquals(HttpStatusCode.OK, response.status())
-            assertNotNull(response.content)
-            assertEquals(listOf("""{"id":1,"title":"Hello, World!","unicode":"$uc"}"""), response.content!!.lines())
+        client.get("/map") {
+            header(HttpHeaders.Accept, "application/json")
+        }.let { response ->
+            assertEquals(HttpStatusCode.OK, response.status)
+            assertEquals("""{"id":1,"title":"Hello, World!","unicode":"$uc"}""", response.bodyAsText())
             val contentTypeText = assertNotNull(response.headers[HttpHeaders.ContentType])
             assertEquals(ContentType.Application.Json, ContentType.parse(contentTypeText))
```

Comment:
CI fails here. Investigating...

Reply:
Okay, I've changed the expected result for now. Any clues on why the actual result has changed with the new DSL are appreciated.